app/queries/search_results_query.rb

```diff
@@ -33,4 +34,23 @@ def filter_adwords_url_contains(word)
       @search_results
     end
   end
+
+  def filter_urls_contains_at_least(character, match_count = 1)
```

Comment:
For consistency, the parameter could be named `word` 💡 It is okay to have your own take of the requirements. You need to build the feature in a way that makes sense.